tests/components/backup/test_http.py

```diff
@@ -65,19 +64,21 @@ async def test_downloading_remote_backup(
     hass_client: ClientSessionGenerator,
 ) -> None:
     """Test downloading a remote backup."""
-    await setup_backup_integration(
-        hass, backups={"test.test": [TEST_BACKUP_ABC123]}, remote_agents=["test"]
+
+    async def download_backup(backup_id: str, **kwargs: Any) -> AsyncIterator[bytes]:
+        """Mock download."""
+        return aiter_from_iter((b"backup data",))
```

Comment:
This is used in two tests, we could add it to the mock returned by `mock_backup_agent` (the mock's `async_download_backup` now just raises `BackupNotFound`)